WeatherApp/ViewControllers/WeatherInfoViewController.swift

```diff
@@ -0,0 +1,171 @@
+    var currentLocationLongitude: Double = 0.0
+    
+    var weatherInfoData: WeatherInfoData?
+    let locationManager = CLLocationManager()
+    
+    @IBOutlet weak var cityName: UILabel!
+    @IBOutlet weak var temperatureLabel: UILabel!
+    @IBOutlet weak var minTemp: UILabel!
+    @IBOutlet weak var maxTemp: UILabel!
+    @IBOutlet weak var descriptionLabel: UILabel!
+    @IBOutlet weak var descriptionIcon: UIImageView!
+    @IBOutlet weak var feelsLikeLabel: UILabel!
+    @IBOutlet weak var weatherInfoTable: UITableView!
+    @IBOutlet weak var weatherInfoViews: UIView!
+    
+    @IBAction func currentLocationButtonTapped(_ sender: UIButton) {
+        fetchWeatherData(latitude: currentLocationLatitude, longitude: currentLocationLongitude)
+    }
+    
+    @IBAction func searchLocationButtonTapped(_ sender: UIButton) {
+        
+        let locationVC = storyboard?.instantiateViewController(withIdentifier: "LocationsViewController") as! SearchCityViewController
+        locationVC.delegate = self
+        navigationController?.pushViewController(locationVC, animated: true)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        getCurrentLocation()
+        registerCustomWeatherCell()
+    }
+    
+    override func viewDidLayoutSubviews() {
+        super.viewDidLayoutSubviews()
+        weatherInfoViews.layer.cornerRadius = weatherInfoViews.frame.size.height / 10
+        weatherInfoViews.layer.masksToBounds = true
+        weatherInfoViews.layer.borderWidth = 0
+    }
+    
+    func registerCustomWeatherCell() {
+        let nib = UINib(nibName: "WeatherInfoCell", bundle: nil)
+        weatherInfoTable.register(nib, forCellReuseIdentifier: "weatherInfoCell")
+    }
+    
+    func getCurrentLocation() {
+        self.locationManager.delegate = self
+        self.locationManager.desiredAccuracy = kCLLocationAccuracyBest
+        self.locationManager.requestWhenInUseAuthorization()
+        self.locationManager.requestLocation()
+    }
+    
+    func fetchWeatherData(latitude: Double, longitude: Double) {
+        let networkManager  = NetworkManager()
+        networkManager.fetchWeatherData(latitude: latitude, longitude: longitude, completionHandler: { response in
+            self.weatherInfoData = response
+            DispatchQueue.main.async  {
+                self.updateWeatherDataInView()
+            }
+        })
+    }
+    
+    func updateWeatherDataInView() {
+        if let weatherInfoData = weatherInfoData {
+            cityName.text = "\(weatherInfoData.name)"
+            self.temperatureLabel.text = "\(Int(weatherInfoData.main.temp))ºC"
+            self.descriptionLabel.text = "\(weatherInfoData.weather[0].description)"
+            let iconName = weatherInfoData.weather[0].icon
+            self.descriptionIcon.image = UIImage(named: iconName)
+            self.feelsLikeLabel.text = "Feels like \(Int(weatherInfoData.main.feels_like))ºC"
```

Comment:
Any reason to convert it to int?

Reply:
> Any reason to convert it to int?

The api's temperature gives a double value, so to get an integer value I converted it to int.